src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -29,58 +29,110 @@ namespace Microsoft.Sarif.Viewer.ErrorList
+
+                        if (response == MessageDialogCommand.Yes)
+                        {
+                            // Prompt for a location to save the transformed log.
+                            outputPath = PromptForFileSaveLocation(Resources.SaveTransformedV1Log_DialogTitle, filePath);
+
+                            if (string.IsNullOrEmpty(outputPath))
+                            {
+                                return;
+                            }
+                        }
+
+                        logText = JsonConvert.SerializeObject(log);
+                    }
+                    else if (inputVersion != VersionConstants.StableSarifVersion)
                     {
-                        // Prompt for a location to save the transformed log.
-                        outputPath = PromptForFileSaveLocation(Resources.SaveTransformedV1Log_DialogTitle, filePath);
+                        // It's an older v2 version, so send it through the pre-release compat transformer.
+                        // Ask if they'd like to save the transformed log.
+                        MessageDialogCommand response = PromptToSaveProcessedLog(string.Format(Resources.TransformPrereleaseV2_DialogMessage, VersionConstants.StableSarifVersion));
 
-                        if (string.IsNullOrEmpty(outputPath))
+                        if (response == MessageDialogCommand.Cancel)
                         {
                             return;
                         }
-                    }
 
-                    logText = JsonConvert.SerializeObject(log);
+                        log = PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(logText, Formatting.Indented, out logText);
+
+                        if (response == MessageDialogCommand.Yes)
+                        {
+                            // Prompt for a location to save the transformed log.
+                            outputPath = PromptForFileSaveLocation(Resources.SaveTransformedPrereleaseV2Log_DialogTitle, filePath);
+
+                            if (string.IsNullOrEmpty(outputPath))
+                            {
+                                return;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        // Since we didn't do any pre-processing, we don't need to write to a temp location.
+                        outputPath = filePath;
+                        saveOutputFile = false;
+                    }
                 }
                 else
                 {
-                    // They're opening a v2 log, so send it through the pre-release compat transformer
-                    PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(logText, Formatting.Indented, out logText);
+                    // The version property wasn't found within the first 100 characters.
+                    // Per the spec, it should appear first in the sarifLog object.
```

Comment:
>should [](start = 40, length = 6)

"Should" is the operative word. The spec suggests that so that a consumer can quickly determine the version by sniffing the file. If we're going to slurp the whole file anyway, you might consider JSON-tokenizing it and just finding the `version` property.

Having said that, we need to get this fix in, so I'd be ok with considering this for a follow-on change.

Reply:
OK